src/bin/svg2png/svg2png.cpp

```diff
@@ -25,180 +25,324 @@
 #include <thorvg.h>
 #include <vector>
 #include "lodepng.h"
+#include <dirent.h>
+#include <unistd.h>
+#include <limits.h>
 
 using namespace std;
 
-
 struct PngBuilder
 {
-    void build(const std::string &fileName , const uint32_t width, const uint32_t height, uint32_t *buffer)
+    void build(const string& fileName, const uint32_t width, const uint32_t height, uint32_t* buffer)
     {
-        std::vector<unsigned char> image;
+        unsigned error = 0;
+        //Used ARGB8888 so have to move pixels now
+        vector<unsigned char> image;
         image.resize(width * height * 4);
         for (unsigned y = 0; y < height; y++) {
             for (unsigned x = 0; x < width; x++) {
-                uint32_t n = buffer[ y * width + x ];
+                uint32_t n = buffer[y * width + x];
                 image[4 * width * y + 4 * x + 0] = (n >> 16) & 0xff;
                 image[4 * width * y + 4 * x + 1] = (n >> 8) & 0xff;
                 image[4 * width * y + 4 * x + 2] = n & 0xff;
                 image[4 * width * y + 4 * x + 3] = (n >> 24) & 0xff;
             }
         }
-        unsigned error = lodepng::encode(fileName, image, width, height);
+        error = lodepng::encode(fileName, image, width, height);
```

Comment:
please declare it when it's necessary.
auto error = lodepng::encode(fileName, image, width, height);